### PR TITLE
Use paymentHook for payments/details call

### PR DESCRIPTION
**Pull request by maassenbas**
## Summary

- Updated sharing permissions
- Reuse paymentHook for payments/details

## Tested scenarios

**Fixed issue**:

### CODE REVIEW

File: force-app/main/default/classes/AdyenController.cls [hunk 1]
```diff
@@ -30,6 +30,22 @@ global with sharing class AdyenController {
         };
     }
 
+    public static Map<String, Object> handlePaymentsDetailsCall(String stateData){
+        PaymentsResponse paymentsDetailsResponse = AdyenPaymentsDetails.requestFromStateData(stateData);
+        Map<String, Object> paymentResult = AdyenResponseHandler.handleResponse(paymentsDetailsResponse);
```
Comment (zaiddreakh) [inline force-app/main/default/classes/AdyenController.cls:35]: I think this `paymentResult` is never used?

File: force-app/main/default/classes/AdyenController.cls [hunk 2]
```diff
@@ -46,6 +62,23 @@ global with sharing class AdyenController {
         insert storedPayment;
     }
 
+    public static Map<String, Object> createProcessorInput(String cartId, String stateData, String adyenCall){
+            ccrz__E_Cart__c cart = AdyenUtil.getCartByEncryptedId(cartId);
+            Map<String, Object> inputParams = new Map<String, Object>();
```
Comment (zaiddreakh) [inline force-app/main/default/classes/AdyenController.cls:67]: Map<String, String>?

File: force-app/main/default/classes/AdyenController.cls (hunk 1 above)
Comment (maassenbas) [reply force-app/main/default/classes/AdyenController.cls:35]: You are right, will update it.

File: force-app/main/default/classes/AdyenController.cls (hunk 2 above)
Comment (maassenbas) [reply force-app/main/default/classes/AdyenController.cls:67]: Yes, can indeed also use Map<String, String> here

**Review by zaiddreakh — APPROVED** [review]

**Review by cyattilakiss — APPROVED** [review]

**Pull request merged**